platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/util/PropertiesFileSupport.java

```diff
@@ -0,0 +1,100 @@
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.ui.admin.api.util;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.PropertiesLoader;
+
+/**
+ * Useful for scanning a directory for properties files and aggregating the data into maps and lists.
+ */
+public class PropertiesFileSupport {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PropertiesFileSupport.class);
+
+    private static final String PROPERTIES_EXTENSION = ".properties";
+
+    /**
+     * Deserializes all on-disk properties files in the given directory into a list of maps.
+     *
+     * @param directoryPath the path to the directory of properties files
+     * @return a list of maps holding the values of the properties files in the directory the given
+     * path pointed at, which can be empty if no properties were found.
+     */
+    public List<Map<String, String>> loadPropertiesFilesInDirectory(String directoryPath) {
+        List<Map<String, String>> loadedPropertiesFiles = new ArrayList<>();
+        if (!StringUtils.isEmpty(directoryPath)) {
+            File directory = new File(directoryPath);
+            if (directory.exists()) {
+                File[] propertyFiles = directory.listFiles((dir, name) -> name.endsWith(
+                        PROPERTIES_EXTENSION));
+                if (propertyFiles != null) {
+                    for (File propertyFile : propertyFiles) {
+                        loadedPropertiesFiles.add(loadPropertiesFile(propertyFile));
+                    }
+                } else {
+                    LOGGER.debug("Property load target {} was not a directory", directoryPath);
+                }
+            }
+        }
+        return loadedPropertiesFiles;
+    }
+
+    /**
+     * Deserializes an on-disk properties file into a map.
+     *
+     * @param filePath the path to the file of properties
+     * @return a map of the values in the properties file the provided path pointed at, which can
+     * be empty if no properties were found.
+     */
+    public Map<String, String> loadSinglePropertiesFile(String filePath) {
+        Map<String, String> propertyMap = new HashMap<>();
+        if (!StringUtils.isEmpty(filePath)) {
+            File propertyFile = new File(filePath);
+            if (propertyFile.exists()) {
```

Comment:
❗️ Feels like we could "merge" L75-83 with `loadPropertiesFile()` and do `return loadProperties(propertyFile)` here instead. Thoughts?